### ISSUE TITLE
Suggest suppressing 

### ISSUE THREAD

**Issue by pothos** [koalaman/shellcheck#2524]
#### For bugs
- Rule Id (if any, e.g. SC1000): none
- My shellcheck version (`shellcheck --version` or "online"): 0.7.2
- [ ] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash
set -euo pipefail

while sleep 1; do
  checksum=$(date '+%s' | tail -c 2 | grep -v 7 | md5sum)
  echo "success (${checksum})"
done

```

The wanted code would look like this:

```
checksum=$(date '+%s' | tail -c 2 | { grep -v 7 || true ; } | md5sum)
```

#### Here's what shellcheck currently says:

(no output)

#### Here's what I wanted or expected to see:

Warning that when `-o pipefail` is used any non-zero exit codes of `grep` should be caught explicitly, e.g., with `{ grep ... || true ; }`
